lib/routes/user/add_funds/add_funds_page.dart

```diff
@@ -43,71 +56,106 @@ class AddFundsState extends State<AddFundsPage> {
         _accountSubscription.cancel();
       }
     });
+    _settingsSubscription = widget._accountBloc.accountSettingsStream.listen((settings) {
+      isIpAllowed(settings.moonpayIpCheck);
+    });
+    _moonPaySubscription = _addFundsBloc.moonPayOrderStream.listen((order) {
+      setState(() {
+        _moonPayOrder = order;
+      });
+    });
+    _checkMoonpayOrderExpiration();
+  }
+
+  isIpAllowed(bool ipCheckSetting) async {
```

Comment:
all ip checking and the decision if to show moonpay or not should be in the bloc.
Also the moonpay URL should be generated in the bloc.
I think we can add moonpayURL in AddFundsResponse model and show the moonpay menu only if this field has value.
This way we can encapsulate the ip logic in the bloc without making the UI complex.

Reply:
[Fixed](https://github.com/breez/breezmobile/pull/162/commits/e575a05e69ee9d79c65ba65bc54d6f1b73380a1c)